src/main/java/com/example/MyBatisDemo/services/StudentService.java

```diff
@@ -39,8 +44,26 @@ public StudentDisplayDto addStudent(StudentAdditionDto studentAdditionDto){
         return getStudentById(studentAdditionDto.getId());
     }
 
-    public StudentSubjectsDisplayDto assignSubjectsToStudent(Long id, StudentSubjectsAdditionDto subjectsAdditionDto){
-        studentRepository.assignSubjectsToStudent(id, subjectsAdditionDto.getSubjectIds());
+    public StudentSubjectsDisplayDto assignSubjectsToStudent(Long id, StudentSubjectsAdditionDto subjectsAdditionDto)throws ResourceNotFoundException {
+        List<SubjectEntity> subjectEntities =subjectRepository.findAllSubjects();
+        int flag = 0;
+        if(subjectsAdditionDto != null){
```

Comment:
The code is too verbose. We can use a lot of already available inbuilt methods like list.contains and stream.filter to reduce the size of code.

Reply:
Done